### ISSUE TITLE
Domains without an APEX record fail out on up command

### ISSUE THREAD

**Issue by kylesloan** [kubernetes-incubator/kube-aws#1443]
I am not sure what the command is trying to do here.  I have tried both with foo.com and k8.foo.com and got this error both times.

```
CREATE_FAILED                           APIEndpointDefaultELBRecordSet  "RRSet of type CNAME with DNS name foo.com. is not permitted at apex in zone foo.com."
```

**Comment by kylesloan** [comment]
further troubleshooting, I had to modify

```
dnsName: foo.com
```

to

```
dnsName: k8.foo.com
```
in cluster.yml and got further

**Comment by mumoshu** [comment]
@kylesloan Hey!

That's how DNS and Route53 works. Fortunately AWS has [ALIAS records](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resource-record-sets-choosing-alias-non-alias.html) that technically allows you to point `foo.com` to your k8s api load balancer, but kube-aws doesn't support it today.

If you have a hard requirement to create it directly on the zone apex, I'd recommend you to see this part of the stack template:

https://github.com/kubernetes-incubator/kube-aws/blob/8c15501dcd17f9d597d4c770642864799988d3b9/core/controlplane/config/templates/stack-template.json#L488-L489

Basically, you could enhance kube-aws by adding if-else block here, so that you can use `AliasTarget` instaed of `ResourceRecords`.

See [this AWS doc](https://docs.aws.amazon.com/ja_jp/AWSCloudFormation/latest/UserGuide/aws-properties-route53-aliastarget.html) about `AliasTarget`, and [this documentation](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resource-record-sets-choosing-alias-non-alias.html) about Alias records.
